update_reminder.sh

```diff
@@ -1,10 +1,73 @@
 #!/bin/bash
-MARKT_LAST_CHANGE=$(wget -q -O- https://wo-ist-markt.de/cities/berlin.json | sed -n 's/.*aktualisiert am \(.*\)",/\1/p')
-BERLIN_LAST_CHANGE=$(wget -q -O- https://daten.berlin.de/datensaetze/berliner-wochen-und-tr%C3%B6delm%C3%A4rkte | sed -n '/.*Aktualisiert.*/,$p' | sed -e '/.*end.*/,$d' | sed -n 's/.*<span .*>\(.*\)<\/span.*/\1/p')
-
-if [[ "$MARKT_LAST_CHANGE" != "$BERLIN_LAST_CHANGE" ]]; then
-  echo "Dates don't match!"
-  echo "daten.berlin.de: $BERLIN_LAST_CHANGE"
-  echo "wo-ist-markt.de: $MARKT_LAST_CHANGE"
-  exit 1;
-fi
+
+set -e #exit on error
+
+#Collect errors and fail in the end
+GOT_ERROR=false
+
+#get HTTP code from site and exit 1 iff it is not 200
+function fail_on_non_200_http_status_code {
+  SITE_NAME=$1
+  HEADER=$2
+  HTTP_STATUS_CODE=$(echo "$HEADER" | sed -n 's|.*HTTP/... \([0-9]*\).*|\1|p' )
+  if [[ "$HTTP_STATUS_CODE" != "200" ]]; then
+    echo "$SITE_NAME returned a non 200-code" > /dev/stderr
+    echo "$HTTP_STATUS_CODE" > /dev/stderr
+    echo "" > /dev/stderr
+    GOT_ERROR=true
+  fi
+}
+
+function check_geojson_api_exists {
+  SITE=$1
+  BERLIN_GEOJSON_URL=$(echo "$SITE" | sed -n 's/.*\(https.*all.geojson\).*/\1/p' )
+  if [[ "$BERLIN_GEOJSON_URL" != "" ]]; then
+    echo "daten.berlin.de: GeoJSON-URL not available!" > /dev/stderr
+    echo "" > /dev/stderr
+    GOT_ERROR=true
+  fi
+  echo "$BERLIN_GEOJSON_URL"
+}
+
+function fail_on_data_update {
+  BERLIN_SITE=$1
+  BERLIN_LAST_CHANGE=$(echo "$BERLIN_SITE" | sed -n '/.*Aktualisiert.*/,$p' | sed -e '/.*end.*/,$d' | sed -n 's/.*<span .*>\(.*\)<\/span.*/\1/p')
+
+  MARKT_LAST_CHANGE=$(wget -q -O- https://wo-ist-markt.de/cities/berlin.json | sed -n 's/.*aktualisiert am \(.*\)",/\1/p')
+
+  if [[ "$MARKT_LAST_CHANGE" != "$BERLIN_LAST_CHANGE" ]]; then
+    echo "Dates don't match!"
+    echo "daten.berlin.de: $BERLIN_LAST_CHANGE"
+    echo "wo-ist-markt.de: $MARKT_LAST_CHANGE"
+    echo "" > /dev/stderr
+    GOT_ERROR=true
+  fi
+}
+
+function get_site_and_header {
+  URL="$1"
+  wget --server-response -q -O- "$URL" 2>&1 #Headers are on stderr
```

Comment:
What do you mean by "Headers are on stderr"?

Reply:
bash scripts have two output-channels, standard out (stdout) and standard error (stderr). The normal data from wget goes to stdout, while the response headers go to stderr. To use the headers normally afterwards, I have to redirect stderr to stdout (that's what 2>&1 does). The comment should only say why I redirect stderr to stdout.

Reply:
Thank you for the clarification.